### ISSUE TITLE
ruff (D): adopt some the ignored rules for the docstrings

### ISSUE THREAD

**Issue by johnnv1** [kornia/kornia#3083]
we need an issue to discuss and check which of these rules we may want to have

_Originally posted by @johnnv1 in https://github.com/kornia/kornia/pull/3082#discussion_r1868460841_

The ignored rules are (from https://docs.astral.sh/ruff/rules/#pydocstyle-d):

- 'D100' :  Missing docstring in public module
- 'D101' :  Missing docstring in public class
- 'D102' :  Missing docstring in public method
- 'D103' :  Missing docstring in public function
- 'D104' : Missing docstring in public package
- 'D105' :  Missing docstring in magic method
- 'D107' :  Missing docstring in __init__
- 'D203' :  1 blank line required before class docstring
- 'D204' :  1 blank line required after class docstring
- 'D205' :  1 blank line required between summary line and description
- 'D213' :  Multi-line docstring summary should start at the second line
- 'D400' :  First line should end with a period
- 'D401' :  First line of docstring should be in imperative mood: "{first_line}"
- 'D404' :  First word of the docstring should not be "This"
- 'D406' :  Section name should end with a newline ("{name}")
- 'D407' :  Missing dashed underline after section ("{name}")
- 'D415' :  First line should end with a period, question mark, or exclamation point
- 'D417' :  Missing argument description in the docstring for {definition}: {name}

maybe we should adopt some default style used by the community (numpy or google) cc @edgarriba @shijianjian @ducha-aiki 

- https://docs.astral.sh/ruff/faq/#does-ruff-support-numpy-or-google-style-docstrings
- https://docs.astral.sh/ruff/formatter/#docstring-formatting

---

Missing rules to be enable after #3088 

- [ ]  'D100'
- [ ]  'D101'
- [ ]  'D102'
- [ ]  'D103
- [ ]  'D104',
- [ ]  'D105'
- [ ]  'D107'
- [ ]  'D417'

**Comment by johnnv1** [comment]
for me, we should adopt the numpy style/convention for docstrings. And from this list of ignored rules, I would only ignore the  D100 and D105 rules... the docstrings on classes or methods or functions should exist, if it's some API that shouldn't be public (aka backbones used internally)  we should just make it "private" by using a `_` into the prefix of the name...


but to do it, is a really big job, which I didn't have time to commit to at the moment

**Comment by edgarriba** [comment]
Sounds good adopting numpy style. Ideally that should be done iteratively. Maybe here or in another ticket draft a small roadmap for this effort so that anyone else can also contribute?

**Comment by johnnv1** [comment]
> Sounds good adopting numpy style. Ideally that should be done iteratively. Maybe here or in another ticket draft a small roadmap for this effort so that anyone else can also contribute?

i think we can do it over this same issue

**Comment by ntjohnson1** [comment]
Note that the numpy doc string style isn't fully supported by ruff yet: https://github.com/astral-sh/ruff/issues/2310 since it doesn't fully support `Parameters` instead of args. Additionally, the current format is MUCH closer to the google doc style. IMO it makes more sense to have the transition to the numpy doc style separate from ruff doc string support (and potentially delayed until it can by better handled by ruff).

**Comment by johnnv1** [comment]
as part of #3088 the google style has been adopted see https://github.com/kornia/kornia/pull/3088#discussion_r1876868330

**Comment by edgarriba** [comment]
@ntjohnson1 do you think that part of this work can be achieved by the end of the month as we have scheduled to release kornia 0.8 ? Which should be really bugfixes over the current version and doc improvements. See: https://github.com/kornia/kornia/wiki/2024#infrastructure

**Comment by ntjohnson1** [comment]
The majority of the work described here is already done. With the open PR the pydocstyle rules are applied/enforced for all existing documentation.

The remaining D10* rules all are related to adding missing documentation. I have limited remaining bandwidth this month, and haven't seen the effort for adding missing doc strings here yet (if it's just some child classes missing docs I can steal content from other classes, if it's classes entirely undocumented it'll take me more time). So I can't confidently commit to fix all the remaining rules by the end of the month.

Happy to let someone else pick it up or I can open incremental PRs per rule like D417.

**Comment by edgarriba** [comment]
Sounds good !

**Comment by harshil0014** [comment]
Hi! I’d like to help adopt these docstring rules. Can I take this on?

**Comment by Ruddellj** [comment]
Hi! I'm interested in contributing to this docstring conversion effort. Since this issue was opened in 2024, I wanted to check:

1. Is this still an active priority for the project?
2. Is anyone currently working on this?
3. Are there any updated guidelines or preferences for the Numpy-style docstring format?

I'd be happy to help convert the docstrings in the geometry module if this is still needed. Thanks!

**Comment by Karthikeya-Thatipamula** [comment]
Hi @johnnv1 ,
I’d like to help with this docstring conversion effort.
I’ve reviewed the recent changes in #3088 and understand the move toward Google-style docstrings for now, with selective adoption of the pydocstyle rules.
I can start by fixing one of the remaining D10* rules (e.g., D103 or D107) in a specific module to keep the changes manageable and iterative.
Could you assign me a module to start with so I can open an incremental PR?

**Comment by edgarriba** [comment]
@Karthikeya-Thatipamula start for the easy ones like color,filter,enhance, etc. Leave for the end augmentation or geometry which are large

**Comment by RagS8i** [comment]
Hi! I’m new to kornia and would like to start contributing by adding docstrings. I was thinking of working on the kornia.filters module, could you please confirm if it’s available or suggest another good starting point with and assign it to me?

**Comment by edgarriba** [comment]
go ahead

**Comment by Big-Marvel** [comment]
Hi @edgarriba @johnnv1 
Is this issue still open? If so, I would like to take this up.

**Comment by soumita94** [comment]
Hi! I’m a beginner contributor preparing for GSoC 2026.
I don’t have prior CV experience, but I’m comfortable with Python and documentation.
I noticed this issue has been open for a while, and I wanted to check if it’s still relevant.
If yes, I’d be happy to help by addressing a small subset of docstring rules (e.g. D400/D401) in a limited scope.

**Comment by edgarriba** [comment]
Coordinate with @Big-Marvel to join efforts on this

**Comment by soumita94** [comment]
Hi @Big-Marvel 
I’m happy to help and align with whatever plan you have.
Since I’m a beginner, I’d prefer to take a small, well-scoped task (for example applying one or two docstring rules like D400/D401 to a single module) to avoid overlap.
Please let me know what would be most helpful.

**Comment by Big-Marvel** [comment]
Hey @soumita94 , I think D400 and D401 have already been implemented, but there are still a few (14 errors) Ruff linting checks failing with D401. You could take those on—they should be quick fixes.

You might also try implementing the D10* rules (which aren't implemented yet) from the list above. I'll handle D104 for now.

**Comment by soumita94** [comment]
Thanks for the clarification!

I can take on the remaining D401 Ruff violations. I’ll start by locating the 14 failing instances and fixing them in a small, focused PR.

I’ll skip D104 as you mentioned you’re handling that. Once the D401 fixes are done, I can also look into one of the D10* rules if that’s still helpful.

I’ll post an update once I have a draft PR ready.

**Comment by soumita94** [comment]
Hi @Big-Marvel,  @edgarriba 
I’ve fixed 6 of the remaining D401 violations in `kornia/feature/adalam/utils.py` and opened a PR for them.

If this looks good, I can take care of the remaining ones in a follow-up PR to keep changes small and easy to review.

**Comment by Big-Marvel** [comment]
Hi @edgarriba , I have raised a PR to add missing package docstrings, fixing the Ruff D104 linting rules. If this looks good to you, I will start working on the remaining rules. Thank you!

**Comment by soumita94** [comment]
There were total 257 D100 violations Since there are many remaining, I started with two small utility modules to validate the approach and style.
If this looks good, I can continue fixing additional D100 issues gradually in follow-up PRs.

**Comment by edgarriba** [comment]
ping here, otherwise i can take myself this

**Comment by Big-Marvel** [comment]
I am working on D101 and D105. 
D103, D104 and D417 are already done.

**Comment by soumita94** [comment]
Thanks for the update, @Big-Marvel!

I’m currently away from my laptop for a short break, so I won’t be able to pick up new fixes immediately. Once I’m back, I can take a look at any remaining D100/D10* items that are still open, if needed.

**Comment by mehru-cyber** [comment]
Hi! I’m going to work on this issue. I’ll enable selected Ruff docstring (D) rules and fix the resulting violations.Is this issue still open? @edgarriba

**Comment by mehru-cyber** [comment]
kindly respond so that i can get started @edgarriba

**Comment by edgarriba** [comment]
Hey @mehru-cyber — welcome, and sorry for the slow reply! Yes, this issue is still open and contributions are very much welcome.

Please go ahead and:
1. Pick specific Ruff `D` rules to enable (see the existing discussion above for which are already done)
2. Fix the violations those rules surface
3. Open a PR referencing this issue

Coordinate with @Big-Marvel to avoid duplicating effort (they were working on D101/D105). If those are done, feel free to tackle the remaining ones.

Looking forward to your PR! 🚀

**Comment by Big-Marvel** [comment]
I will raise a PR for D105, @mehru-cyber you may take up the rest.

**Comment by gautambandil08** [comment]
Hi, @Big-Marvel I’m a contributor interested in helping with this issue.
I see some rules are already being handled by others.
Is there a specific Ruff docstring rule (for example D100/D102/D103/D417) that is still available for me to work on?

**Comment by Big-Marvel** [comment]
Hey @gautambandil08, D103 and D417 are already done, you may take up D100 and D102.

**Comment by Gaspard-Vernin** [comment]
Hello, I would like to contribute to this issue. Can I fix one of the missing rules ? I would like to contribute to bigger issues after this one !

**Comment by passionworkeer** [comment]
建议采用 Google 或 NumPy 风格的 docstring 作为项目标准。可以在 pyproject.toml 中配置 ruff 的 docstring-format 选项。先从较简单的规则开始，如 D400（首行以句号结尾）和 D401（祈使语气），这些容易统一且收益明显。D213 和 D205 对于已有大量代码的项目可能需要逐步迁移。

**Comment by Gaspard-Vernin** [comment]
Great ! I will work on the D400 and the D401 ! Thanks for your answer :)

**Comment by gautambandil08** [comment]
> Hey [@gautambandil08](https://github.com/gautambandil08), D103 and D417 are already done, you may take up D100 and D102.

Hi @Big-Marvel, thanks for confirming.
I’ll work on fixing the **D100 (Missing docstring in public module)** and **D102 (Missing docstring in public method)** rules and open a PR referencing this issue once done.

**Comment by Gaspard-Vernin** [comment]
Hello, as passionworkeer suggest to me, I made a PR for the D400 rules, 2 tests are not passing but I'm not sure they are related with my work... 

<img width="1299" height="265" alt="Image" src="https://github.com/user-attachments/assets/8cfb10bd-0cb5-4a47-b499-16ae8b9f5234" />

**Comment by MOHITPRADHAN35** [comment]
Hi, I’d like to contribute to this issue.
I plan to start with a small subset of docstring fixes to ensure alignment with ruff rules and get feedback before proceeding further. Does that sound good?

**Comment by shijianjian** [comment]
Hi @MOHITPRADHAN35, it is better to fix them all in one PR, I think. You can mark the PR as a draft to check the warnings.

**Comment by Jeevang1-epic** [comment]
Hi team, I would like to help tackle some of the missing docstrings to help close this issue as my first contribution.

**Comment by ducha-aiki** [comment]
Oh, maybe let's not? Right now many of the ruff things are not really meaningful, and we are going to adopt even more? 
"first line, ", 'D400' : First line should end with a period  are useless bullshit

On the other hand -- yes, 
'D100' : Missing docstring in public module
'D101' : Missing docstring in public class
'D102' : Missing docstring in public method
'D103' : Missing docstring in public function
'D104' : Missing docstring in public package
'D105' : Missing docstring in magic method
'D107' : Missing docstring in init
 
Those are important and we should add them

**Comment by shijianjian** [comment]
I think D100/D104 are unnecessary since our filenames shall be expressive enough.

Hi @Jeevang1-epic, you can probably start with D101, D102, and D103.

**Comment by Jeevang1-epic** [comment]
Understood completely. I will strictly ignore the stylistic formatting rules like D400 and focus only on the structural missing docstrings (D101, D102, D103, etc.). I actually just finished patching planar_tracker.py for those specific rules and will submit the PR for it now.

**Comment by varpekalyani** [comment]
Hi, I am a GSoC 2026 applicant and I would like to work on this issue as my first contribution. May I be assigned?

**Comment by gautambandil08** [comment]
Hey @Big-Marvel, the D100 docstrings for all 245 modules are ready for review! (I updated the description above to properly link Fixes #3083).

**Comment by vedbo** [comment]
Hey! I'm interested in helping with the Ruff docstring migration as part of my GSoC 2026 preparation. I see that kornia.geometry and kornia.filters still need updates. I'd love to take those on! My portfolio: https://vedbo.github.io/

**Comment by heckkkerrr** [comment]
Hi team! I am also preparing for GSoC 2026 and would love to help knock out the rest of this issue. Since geometry, filters, tracking, and augmentation are already being handled, I will take on the kornia.losses and kornia.feature modules.

I will strictly follow the guidance above: ignoring the stylistic rules and focusing entirely on adding missing structural docs (D101, D102, and D103). I'll open a draft PR shortly!

**Comment by heckkkerrr** [comment]
Hi @edgarriba, I've implemented the structural docstrings for kornia.losses.dice to address D101, D102, and D103. PR is live here: #3702. I'd love to be assigned to this module if possible!
